File: pgm/ABC_Java/ABC_261_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final int L1 = Integer.parseInt (arr[0]);
        final int R1 = Integer.parseInt (arr[1]);
        final int L2 = Integer.parseInt (arr[2]);
        final int R2 = Integer.parseInt (arr[3]);
        System.out.println (
            Math.max (Math.min (R1, R2) - Math.max (L1, L2), 0));
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_261_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.Scanner
// import java.util.StringTokenizer

import kotlin.math.*

const val space: String = " "
const val zero: Int = 0

fun main() {
    val (L1: Int, R1: Int, L2: Int, R2: Int) = readln().split(space).map { it.toInt() }
    println((min(R1, R2) - max(L1, L2)).coerceAtLeast(zero).toString())
    kotlin.system.exitProcess(0)
}